src/indexed_crate.rs

```diff
@@ -602,6 +606,10 @@ impl<'a> IndexedCrate<'a> {
         value.fn_owner_index = Some(fn_owner_index);
         value.export_name_index = Some(build_export_name_index(&crate_.index));
 
+        // TODO: Could we use pub_item_kind_index to construct this index?
+        // I expect it would be faster, but would it lose information?
+        value.variant_name_index = Some(build_variant_name_index(&crate_.index));
```

Comment:
It would lose information because it would only include `pub enum` items instead of all `enum` items.

That _might_ be acceptable? It's a question of:
- Is the construction time meaningfully different?
- Are there queries we'd want to support that would be unable to use the narrower index?
- Would it be worth the increase in code complexity required to only apply the index when we know the enum must be `pub`, either directly via a filter on `visibility_limit` that requires it to be `pub`, or via a mandatory `importable_path` edge? Or if we only apply the filter in one of those two cases, if the missed optimization is worth missing to speed up the index construction?

My gut feeling is "not worth it" but would be open to be convinced otherwise.

Reply:
Construction time doesn't appear to be significantly different, so it doesn't appear to be worth making the change.